Arelle/Arelle
# Duplicate fact detection

#### Reason for change
Resolves #1001
Continuing from previous closed PR #1015 
Note: Not mentioned in the original issue was an "incomplete" parameter that OIM requires. The use case of retrieving any duplicates that are *not* complete was not captured in the original issue.

#### Description of change
Implement `ValidateDuplicateFacts` for retrieving duplicate facts of various "types" within an instance.
Add `--validateDuplicateFacts` CLI argument to trigger logging of duplicate facts
Add `Tools > Validation > Warn on duplicate facts` menu to trigger logging of duplicate facts from GUI
Add `fact_deduplication.py` integration test script to test duplicate fact detection (and eventually deduplication as well: #1002)
Replace `loadFromOIM` implementation of duplicate fact detection with internal implementation.

#### Steps to Test
`fact_deduplication.py` integration test script tests this via CLI fairly thoroughly.
New GUI option should be tested manually (ensure option defaults to "None" on first launch)

**review**:
@Arelle/arelle
